storage/bucket.go

```diff
@@ -1373,7 +1373,7 @@ func toRetentionPolicy(rp *raw.BucketRetentionPolicy) (*RetentionPolicy, error)
 }
 
 func toRetentionPolicyFromProto(rp *storagepb.Bucket_RetentionPolicy) *RetentionPolicy {
-	if rp == nil {
+	if rp == nil || rp.EffectiveTime.AsTime().Unix() == 0 {
```

Comment:
```suggestion
	if rp == nil || rp.GetEffectiveTime().AsTime().Unix() == 0 {
```

Always use the accessor methods when reading a proto field. It protects us from _some_ breaking changes and also handles the case of the receiver message being `nil` to avoid panics.

Reply:
done